### PR TITLE
Tdd project

### PR DESCRIPTION
**Pull request by ikoote1**
Changes made;

1. Created a class called Solver.
2. Created a method called factorial that takes one argument, an integer N, and returns the [factorial](https://en.wikipedia.org/wiki/Factorial) for that number. The factorial is the multiplication of all integers from 1 to N and has the special case that the factorial of 0 is 1. This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception.
3. Created a method called reverse that takes one argument, a string word, and returns word reversed (e.g. if word is "hello" it returns "olleh").
4. Created a method called fizzbuzz that takes one argument, an integer N, and returns a string. The returned string is constructed following these rules:

- When N is divisible by 3, return "fizz".
- When N is divisible by 5, return "buzz".
- When N is divisible by 3 and 5, return "fizzbuzz".
- Any other case, return N as a string (e.g. say N is 7 then return "7").
5. Started with tests for all of the methods described above by following TDD.
6. Opened a pull request

Thanks for reviewing my code.

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+require_relative '../solver'
+
+describe 'Solver' do
+  let(:solver) { Solver.new }
+
+  describe 'Factorial' do
+    it 'returns the factorial of a number' do
+      expect(solver.factorial(5)).to eq(120)
+    end
+  end
```
Comment (Ismailco) [inline spec/solver_spec.rb]: - [x] Kindly make sure to test the Negative value which raises an exception as an expectation

Comment (Ismailco) [inline spec/solver_spec.rb]: - [x] Also kindly make sure to test the special case when the input is `0`, the result should be `1`

File: spec/solver_spec.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    it 'Returns fizzbuzz if number is divisble by both 5 and 3' do
+      expect(solver.fizzbuzz(15)).to eq('fizzbuzz')
+    end
+    it 'Returns fizz if number is divisble by 3' do
+      expect(solver.fizzbuzz(9)).to eq('fizz')
+    end
+    it 'Returns buzz if number is divisble by 5' do
+      expect(solver.fizzbuzz(25)).to eq('buzz')
+    end
+  end
```
Comment (Ismailco) [inline spec/solver_spec.rb]: - [x] Kindly consider adding one other case for your tests, the case should be when the number is not devisable by `3` nor by `5` this should return the number 
e.g `7` should return `7`

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @ikoote1, @MuhammadUsmanAslam, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ Uses TDD
✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mwanawabangona — APPROVED** [review]
Hi @ikoote1,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  


## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits
*   Good job implementing the requested changes!! :clap: :clap: :clap:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**